src/components/periodic-table-page/PeriodicTablePage.test.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from "react";
+import {
+  render,
+  screen,
+  waitForElementToBeRemoved,
+} from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import PeriodicTablePage from "./PeriodicTablePage";
+import { Router } from "react-router-dom";
+import { createMemoryHistory } from "history";
+import "hammerjs";
+
+test("should render periodic table and able to see details of element", async () => {
+  const history = createMemoryHistory({
+    initialEntries: ["/periodic-table"],
+  });
+  render(
+    <Router history={history}>
+      <PeriodicTablePage />
+    </Router>
+  );
+
+  await waitForElementToBeRemoved(
+    () => screen.queryAllByLabelText(/loading/i),
+    { timeout: 4000 }
+  );
+
+  userEvent.click(screen.getAllByRole("button")[1]);
```

Comment:
Could this query be done using `name`? 
Something like `userEvent.click(screen.getByRole("button", { name: "1 H Hydrogen" }));`

Of course, hydrogen will always be the first element but the test might break if other buttons are added 😅

Reply:
Indeed. I've made it more specific now